python/pyarrow/includes/libarrow_dataset.pxd

```diff
@@ -194,7 +194,11 @@ cdef extern from "arrow/dataset/api.h" namespace "arrow::dataset" nogil:
         const c_string& path() const
         const shared_ptr[CFileSystem]& filesystem() const
         const shared_ptr[CBuffer]& buffer() const
-        CFileSource(c_string path, shared_ptr[CFileSystem] filesystem)
+        # HACK: Cython can't handle all the overloads so don't declare them.
```

Comment:
I'm curious, did you report a bug to Cython?

Reply:
No. I can try to assemble a minimal reproducer

Reply:
Even without a minimal reproducer, if you just post the relevant snippets in a bug report (declarations then point of use), it may be enough for the Cython team to understand and fix the issue.